File: panaderia_api/src/schemas/ai.py
from typing import Any
from pydantic import BaseModel


class ChatRequest(BaseModel):
    message: str
    conversation_id: str | None = None


class ToolSource(BaseModel):
    tool: str
    result: dict[str, Any]


class ChatResponse(BaseModel):
    reply: str
    conversation_id: str
    sources: list[ToolSource]

File: panaderia_web/src/api/ai.ts
import { apiClient } from './client'

export interface ChatRequest {
  message: string
  conversation_id: string | null
}

export interface ToolSource {
  tool: string
  result: Record<string, unknown>
}

export interface ChatResponse {
  reply: string
  conversation_id: string
  sources: ToolSource[]
}

export async function sendChatMessage(body: ChatRequest): Promise<ChatResponse> {
  const { data } = await apiClient.post<ChatResponse>('/api/v1/ai/chat', body)
  return data
}
